src/KubernetesClient/Informers/Cache/SimpleCache.cs

```diff
@@ -0,0 +1,199 @@
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace k8s.Informers.Cache
+{
+    public class SimpleCache<TKey, TResource> : ICache<TKey, TResource>, ICacheSnapshot<TKey,TResource>
+    {
+        private readonly IDictionary<TKey, TResource> _items;
+
+        public SimpleCache()
+        {
+            _items = new Dictionary<TKey, TResource>();
+        }
+
+        public SimpleCache(IDictionary<TKey, TResource> items, long version)
+        {
+            Version = version;
+            _items = new Dictionary<TKey, TResource>(items);
+        }
+
+        public void Reset(IDictionary<TKey, TResource> newValues)
+        {
+            lock (SyncRoot)
+            {
+                _items.Clear();
+                foreach (var item in newValues)
+                {
+                    _items.Add(item.Key, item.Value);
+                }
+
+            }
+        }
+
+        public object SyncRoot { get; } = new object();
```

Comment:
Move this to the beginning of the object, document that it is used as a lock.

Reply:
Also, why is it accessble if it is just used a lock?

Reply:
I originally introduced SyncRoot as ways to "pause" cache updates while doing some UOW but then moved towards Version & Snapshot based interface. It's not needed anymore, I'll remove